big-red-shuttle/Controllers/DriverViewController.swift

```diff
@@ -1,17 +1,127 @@
 import UIKit
 
-class DriverViewController: UIViewController {
+class DriverViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    var tableView: UITableView!
+    let data = UserDefaults.standard
+    var loginImage: UIImage?
+    var isLoggingOn = false
+    var textFieldText: String = ""
+    var loginSuccessful: Bool = false
     
-    var dismissButton: UIButton!
-    var titleLabel: UILabel!
-    var descriptionTextView: UITextView!
-    var entryTextField: UITextField!
-
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        navigationController?.navigationBar.topItem?.title = "Settings"
+        let closeButton = UIBarButtonItem(barButtonSystemItem: .stop, target: self, action: #selector(dismissView))
+        closeButton.tintColor = .darkGray
+        navigationController?.navigationBar.topItem?.rightBarButtonItem = closeButton
+        view.backgroundColor = .brslightgrey
+        let tableViewFrame = CGRect(x: 0, y: 0, width: view.bounds.width, height: view.bounds.height)
+        tableView = UITableView(frame: tableViewFrame, style: .grouped)
+        tableView.isScrollEnabled = false
+        tableView.dataSource = self
+        tableView.delegate = self
+        view.addSubview(tableView)
+    }
+    
+    
+    func dismissView() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return "Logging Login"
+    }
+    
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 2
+    }
 
-        view.backgroundColor = .darkGray
     
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = UITableViewCell(style: .default, reuseIdentifier: "cell")
+        if indexPath.row == 0 {
+            
+            let textField = UITextField(frame: CGRect(x: cell.bounds.width/2 + 20, y: 0, width: cell.bounds.width/2, height: cell.bounds.height))
+            cell.textLabel?.text = "Authentication Key"
+
+            if loginImage != nil {
+            cell.imageView?.image = loginImage
+            textField.text = textFieldText
+            }
+            textField.placeholder = "Insert Key"
+            textField.textAlignment = .right
+            textField.returnKeyType = .go
+            textField.isSecureTextEntry = true
+            textField.addTarget(self, action: #selector(textFieldDidChange), for: .editingDidEndOnExit)
+        cell.accessoryView = textField
+        } else {
+            if loginSuccessful {
+                cell.isUserInteractionEnabled = true
+                cell.textLabel?.textColor = .black
+            } else {
+                cell.isUserInteractionEnabled = false
+                cell.textLabel?.textColor = .lightGray
+                cell.textLabel?.alpha = 0.3
+            }
```

Comment:
lines 63-69 condense to:
`cell.isUserInteractionEnabled = loginSuccessful`
`cell.textLabel?.textColor = loginSuccessful ? . black : .lightGray`
`cell.textLabel?.alpha = loginSuccessful ? 1.0 : 0.3`